app/helpers/date_range_helper.rb

```diff
@@ -1,45 +1,47 @@
 # Encapsulates methods used on the Dashboard that need some business logic
 module DateRangeHelper
-  def display_interval
-    selected_interval.humanize.downcase
+  def date_range_params
+    params.dig(:filters, :date_range).presence || this_year
   end
 
-  def filter_intervals
-    [
-      %w(Today today),
-      %w(Yesterday yesterday),
-      ["This Week to date", "this_week"],
-      ["This Month to date", "this_month"],
-      ["Last Month", "last_month"],
-      ["This Year to date", "this_year"],
-      ["Last Year", "last_year"],
-      ["All Time", "all_time"],
-    ]
+  def date_range_label
+    case (params.dig(:filters, :date_range_label).presence || "this year").downcase
+    when "yesterday"
+      "yesterday"
+    when "last 7 days"
+      "over the last week"
+    when "last 30 days"
+      "over the 30 days"
+    when "this month"
+      "this month, so far"
+    when "last month"
+      "during the last month"
+    else
+      selected_range_described
+    end
+  end
+
+  def this_year
+    "01/01/#{Time.zone.today.year} - 12/31/#{Time.zone.today.year}"
   end
 
   def selected_interval
-    params.dig(:filters, :interval) || "this_year"
+    date_range_params.split(" - ").map { |d| Date.strptime(d, "%m/%d/%Y") }
   end
 
   def selected_range
-    now = Time.zone.now.end_of_day
-    case selected_interval
-    when "today"
-      now.beginning_of_day..now
-    when "yesterday"
-      (now - 1.day).beginning_of_day..(now - 1.day).end_of_day
-    when "this_week"
-      now.beginning_of_week..now
-    when "this_month"
-      now.beginning_of_month..now
-    when "last_month"
-      (now - 1.month).beginning_of_month..(now - 1.month).end_of_month
-    when "this_year"
-      now.beginning_of_year..now.end_of_year
-    when "last_year"
-      (now - 1.year).beginning_of_year..(now - 1.year).end_of_year
+    start_date, end_date = selected_interval
+    start_date..end_date
+  end
+
+  def selected_range_described
+    start_date, end_date = selected_interval
+    if start_date == Time.zone.today
+      return ""
+    elsif end_date == Time.zone.today
+      return "since #{start_date}"
     else
-      Time.zone.local(2017, 1, 1, 0, 0, 0)..Time.zone.local(2037, 1, 1, 0, 0, 0)
+      return "during the period #{start_date.to_s(:short)} to #{end_date.to_s(:short)}"
```

Comment:
Is there a reason we need an explicit `return` here?

Reply:
Just bein' explicit -- I know ruby would just fall through to a return, but I didn't want to take chances if it gets edited later.